File: src/main/java/io/protobase/f7/formulas/math/PI.java
package io.protobase.f7.formulas.math;

import io.protobase.f7.formulas.AbstractFormula;
import io.protobase.f7.formulas.FormulaName;
import io.protobase.f7.models.GridColumnRowKey;

import java.util.function.BiFunction;
import java.util.function.Function;

public class PI extends AbstractFormula {
  public static FormulaName NAME = FormulaName.PI;
  public static PI SELF = new PI();

  public PI() {
    super();
  }

  public PI(Function<Object, Object> lookup, BiFunction<GridColumnRowKey, Object, Object> collateralLookup) {
    super(lookup, collateralLookup);
  }

  @Override
  public Object internal(GridColumnRowKey origin, Object... values) {
    checkLength(values.length, 0, NAME);
    return Math.PI;
  }
}

File: src/main/js/formulas/math/PI.ts
import { SheetColumnRowKey } from "../../models/common/SheetColumnRowKey";
import { Complex } from "../../models/common/Types";
import { AbstractFormula } from "../AbstractFormula";
import { FormulaName } from "../FormulaName";

export class PI extends AbstractFormula {
  static SELF: PI = new PI();
  NAME = FormulaName.PI;

  internal(origin: SheetColumnRowKey, ...values: Array<Complex>) {
    AbstractFormula.checkLength(values.length, 0, this.NAME);
    return Math.PI;
  }
}
